draft-ietf-anima-brski-prm.md

```diff
@@ -1103,118 +1128,123 @@ pledgeenrollrequesttrigger = {
+
+* `created-on`: SHALL contain the current date and time as standard date/time string as defined in {{Section 5.6 of !RFC3339}};
+  if the pledge does not have synchronized time, it SHALL use the `created-on` value from the BRSKI-PRM Agent-Signed Data, received in the trigger to create the PVR and SHOULD advance that value to reflect the PER creation time
 
-# Example: Decoded "JWS Protected Header" Representation
-  in JSON Syntax
+The new protected Header Parameter `created-on` is introduced to reflect freshness of the PER.
+It allows the registrar to verify the timely correlation between the PER artifact and previous exchanges, i.e., `created-on` of PER >= `created-on` of PVR >= `created-on` of PVR trigger.
+The registrar MAY consider to ignore any but the newest PER artifact from the same pledge in the case the registrar has at any point in time more than one pending PER from the pledge.
+
+{{per_header}} below shows an example for this JWS Protected Header:
+
+~~~~
 {
   "alg": "ES256",
   "x5c": [
     "base64encodedvalue==",
     "base64encodedvalue=="
   ],
-  "crit":["created-on"],
+  "crit": ["created-on"],
   "created-on": "2022-09-13T00:00:02.000Z"
 }
 ~~~~
-{: #per_example title='Representation of PER' artwork-align="left"}
+{: #per_header title='JWS Protected Header Example within PER' artwork-align="left"}
+
+#### JWS Signature
+
+The pledge MUST sign the PER artifact using its IDevID credential.
+The JWS Signature is generated over the JWS Protected Header and the JWS Payload as described in {{Section 5.1 of RFC7515}}.
 
-With the collected PVR and PER, the Registrar-Agent starts the interaction with the domain registrar.
+While BRSKI-PRM targets the initial enrollment, re-enrollment can be supported in a similar way.
+In this case, the pledge MAY use its current LDevID credential instead of its IDevID credential to sign the PER artifact.
 
-The new protected header field "created-on" is introduced to reflect freshness of the PER.
-The field is marked critical "crit" to ensure that it must be understood and validated by the receiver (here the domain registrar) according to {{Section 4.1.11 of RFC7515}}.
-It allows the registrar to verify the timely correlation between the PER and previously exchanged messages, i.e., created-on of PER >= created-on of PVR >= created-on of PVR trigger.
-If the pledge does not have synchronized time, it used the created-on time from the agent-signed-data during the creation of the PVR and should advance that value for use in PER creation.
-The registrar MAY consider to ignore any but the newest PER from the same pledge in the case the registrar has at any point in time more than one pending PER from the pledge.
+TODO: created-on for pledge that does not have synchronized time -- cannot use outdated tPVR created-on, as advance the value does not work for long timespans (reboots etc. without RTC).
 
```

Comment:
It was intended that the tPVR date is used as a base, but advanced based on a local clock. So the assumption is to have at least a clock, which may not be synchronized. It is understood, that the advancement of the clock may be interrupted due to reboot or other events. Therefore, it may be necessary the pledge cyclically stores the advancement value until the PER is sent.

Reply:
That's a lot of additional requirements that are not mentioned in the draft.
I would like to see an instruction, which must be the minimum base value for providing `created-on` in a re-enrollment PER -- analog to the `created-on` from the tPVR for initial enrollment, which however cannot be used for re-enrollment, as it is way to old. It maybe could be the "Not Before" date-time from the pledge LDevID cert.

Reply:
We actually did not discuss interruptions in the enrollment process in that way. So I was trying to  come up with an approach similar to the "advancement based on local clock" above. 

With the sentence in 1221 we just wanted to open the door for re-enrollment without providing specifics. The initial synchronization should be during the initial onboarding, based on the tPVR as base, afterwards again the expectation would be that the clock is advanced based on that inital (rough) time. If the pledge looses the time, then it may also not be able to verify received certificates in an application context. 
Based on that we may add a statement that a pledge without active time synchronization SHALL treat the received time stamp in the tPVR advance it based on the local clock in a persistent manner. 

Not that BRSKI used to have a time bootstrapping description in the drafts around version 18, it was removed later on.

Reply:
Again, this is about RE-enrollment, where there is no fresh tPVR timestamp that can be used as base.
What minimum base time SHOULD be used?
For now, I only added fuzzy text about advancing over a longer period, "which also requires persisting the local clock advancements across reboots".

Reply:
I know it is targeting re-enrollment, therefore I put the statement regarding clock advancement by the pledge.

I think that statement is fine. That is what I was looking for. Even it is vague. As said, also BRSKI has dropped the hints on how to bootstrap time and I think it is fair to state it that way. Also, advancing the clock based on some synchronization is necessary to be able to validate certificates in general. If the pledge would loose the time during every reboot, it likely would not be able to validate certificates, even if already onboarded due to the valid-from date in certificates, which is likely later than September 1970.

Reply:
Good, let's leave it at that.